### ISSUE TITLE
Missing archive and backup files in S3 - 503 Slow Down erros

### ISSUE THREAD

**Issue by aamallmann** [pgbackrest/pgbackrest#1290]
1. pgBackRest version:
**2.31**

2. PostgreSQL version:
**10.15**

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each:
**CentOS 7**

4. Did you install pgBackRest from source or from a package?
**package**

5. Describe the issue:
Some files are not successfully saved into S3 due 503 Slow Down Erros, and pgBackRest processes are not taking this as a failure.

### example from archive
[archive.log](https://github.com/pgbackrest/pgbackrest/files/5931790/archive.log)
```
archive-push:async
2021-02-04 00:12:01.770 P00   INFO: push 1 WAL file(s) to archive: 0000000200008591000000D6
2021-02-04 00:12:03.776 P00   INFO: archive-push:async command end: completed successfully (2007ms)

archive-push:local
2021-02-04 00:12:03.674 P00   INFO: archive-push:local command end: completed successfully (1856ms)
```
but backup fails due `check archive` error

```
2021-02-04 09:05:48.784 P00  ERROR: [082]: WAL segment 0000000200008591000000D6 was not archived before the 1800000ms timeout
```

### example from backup/restore
```
2021-02-01 00:12:28.305 P06   INFO: backup file /data/10/pgdata/base/47328763/690123125.55 (1GB, 0%) checksum aa32b18b847bc4ca55c6e03bef37359f731fdb4d
2021-02-01 15:26:30.168 P00   INFO: check archive for segment(s) 000000030000D553000000DD:000000030000D565000000DF
2021-02-01 15:29:46.881 P00   INFO: new backup label = 20210201-001121F
2021-02-01 15:29:48.736 P00   INFO: backup command end: completed successfully (55125122ms)
```
(takes a long time, but finishes "successfully")

error from restore
```
2021-02-03 16:40:28.450 P00  ERROR: [055]: raised from local-49 protocol: unable to open ‘/pgbackrest/<REDACTED_STANZA_NAME>/backup/<REDACTED_STANZA_NAME>/20210201-001121F/pg_data/base/47328763/690123125.55.lz4’: No such file or directory
```
[pgbackrest.conf.txt](https://github.com/pgbackrest/pgbackrest/files/5931845/pgbackrest.conf.txt)

**Comment by dwsteele** [comment]
Well, that certainly doesn't look good. I can see the retries but the debug log level is not sufficient to see what is going on with S3 in the `local` process. Can you capture this event in the `local` log at trace level?

**Comment by aamallmann** [comment]
I'll try to reproduce this scenario in a development environment.
This examples happened in production, we were starting all backups at the same time 00:11, and between 00:12 and 00:13 we were taking several 5xx errors (according to S3 metrics)
I changed the time of the backups to not start all at once, and today I didn't have any archives lost during the backup (don't know about backup files)
Returning as it was before is not feasible, but I will do my best to reproduce this error in development.

**Comment by dwsteele** [comment]
> This examples happened in production, we were starting all backups at the same time 00:11, and between 00:12 and 00:13 we were taking several 5xx errors (according to S3 metrics)

It can take S3 a while to scale a prefix so this is not too surprising. But we do back off and there is no evidence of HTTP errors -- yet the file does not appear. 

> Returning as it was before is not feasible

Understood. If you can reproduce in dev that would be fantastic.

I spent some time adding extra 503 errors into our tests and could not find any issues. It doesn't mean there aren't any but I was not able to find them.

One thing we *could* do is simply HEAD the file after the upload and make sure it really exists. We have a job-level retry which takes care of errors that span HTTP requests, which would kick in and retry.

**Comment by dwsteele** [comment]
There is a good change this is related to the fix in #1438. Closing since there has been no further information and we believe the issue to be fixed.
